internal/ioextensions/io.go

```diff
@@ -64,3 +64,24 @@ func CreateFileWith(filePath string, content io.Reader) error {
 	_, err = utility.FastCopy(file, content)
 	return err
 }
+
+type MultiCloser struct {
+	closers []io.Closer
+}
+
+func NewMultiCloser(closers []io.Closer) *MultiCloser {
+	return &MultiCloser{
+		closers: closers,
+	}
+}
+func (m *MultiCloser) Close() error {
+	var err error
+	for _, c := range m.closers {
+		// still call Close on each, even if one returns an error
```

Comment:
hm, it will be better to save all error messages. You can combine it to a new error

Reply:
Fixed in https://github.com/wal-g/wal-g/pull/736/commits/4efeef112992fd0234f6363143c9e6cbc4545304